### ISSUE TITLE
IME doesn't send composition events to browsers on Windows

### ISSUE THREAD

**Issue by petrsimon** [mugli/Avro-Keyboard#14]
Hi,
I was testing Avro IME both on Windows and Linux and on Windows the IME doesn't send any composition events, while ibus-avro does. The missing composition events break text engines which handle input events on a lower level. I'm using the phonetic IME.
Thanks

**Comment by sarim** [comment]
Yes, windows version isn't implemented as an IME. Can you be more specific what real world problem you are facing because of that?

**Comment by petrsimon** [comment]
Thanks Sarim, that explains it then. 

The issue seems to be caused by the fact that Avro is somehow trying to replace the characters is has already sent to the platform during it's internal composition phase. But the text engine I'm working on, is receiving input in a hidden `<input>` element, removes what's been inserted and moves it to another element which serves as a rich text display. 

I can handle regular IME for CJK languages quite well and obviously non-standard inputs like Avro don't work. I hope that explains a bit the issue I'm facing. 

I've received complains from Bangla users that e.g. typing `sthi` produces স্তথি instead of স্থি.

**Comment by mugli** [comment]
This one should be improved when [this "hack"](https://github.com/mugli/Avro-Keyboard/pull/43/files#diff-6ec544f29f290a6e91d5309c0d28322f4a1926f08eb4123957a6981211f40783R353-R377) lands in the next release. But the options for fixing this are limited based on the way Avro Keyboard processes and sends keystrokes in Windows, unfortunately. This project came before Text Services Framework became a thing in Windows. :)

**Comment by mugli** [comment]
Closing as fixed (https://github.com/mugli/Avro-Keyboard/pull/54, https://github.com/mugli/Avro-Keyboard/pull/76)
